src/store/hooks/useAuth.tsx

```diff
@@ -0,0 +1,212 @@
+
+interface UserWrapped {
+  user: UserOptions,
+  determiningStatus: boolean,
+  isAdmin: boolean,
+  adminVerification:boolean
+}
+
+// Defines default AuthContextValue
+interface AuthContextValue {
+  user: UserWrapped;
+  logout: () => void;
+  verifyAdminStatus: (firebaseUser: UserOptions) => Promise<boolean>;
+  }
+
+// Initializes AuthContext
+const AuthContext = createContext<AuthContextValue>({
+  user: {
+    user: null,
+    determiningStatus: false,
+    isAdmin: false,
+    adminVerification: false,
+  },
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  logout: () => {},
+  verifyAdminStatus: () => Promise.resolve(false),
+
+});
+
+// Firebase auth context
+export const useAuth = () => useContext(AuthContext);
+
+// Defines the functions that are exposed in this hook.
+export function AuthProvider({ children, globalConfig } : { children: ReactNode, globalConfig: GlobalConfig }) {
+  const [user, setUser] = useState<UserWrapped>(
+    {
+      user: null,
+      determiningStatus: true,
+      isAdmin: false,
+      adminVerification: false,
+    },
+  );
+
+  // Default non-user when loading
+  const loadingNullUser : UserWrapped = {
+    user: null,
+    determiningStatus: true,
+    isAdmin: false,
+    adminVerification: false,
+  };
+
+  // Default non-user when not loading
+  const nonLoadingNullUser : UserWrapped = {
+    user: null,
+    determiningStatus: false,
+    isAdmin: false,
+    adminVerification: false,
+  };
+
+  // Non-auth User
+  const nonAuthUser : UserWrapped = {
+    user: {
+      name: 'fakeName',
+      email: 'fakeEmail@fake.com',
+    },
+    determiningStatus: false,
+    isAdmin: true,
+    adminVerification: true,
+  };
+
+  const { storageEngine } = useStorageEngine();
+
+  // Logs the user out by removing the user and navigating to '/login'
+  const logout = async () => {
+    const auth = getAuth();
+    try {
+      await signOut(auth);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      console.error(`There was an issue signing-out the user: ${error.message}`);
+    }
+    setUser(nonLoadingNullUser);
+  };
+
+  function isFirebaseUser(input: UserOptions): input is User {
+    return input !== null && typeof input === 'object' && 'uid' in input;
+  }
+
+  function isLocalStorageUser(input: UserOptions): input is LocalStorageUser {
+    return input !== null && typeof input === 'object' && !('uid' in input);
+  }
+
+  const verifyAdminStatus = async (inputUser: UserOptions) => {
+    if (inputUser === null) {
+      return true;
+    } if (isFirebaseUser(inputUser)) {
+      // Assert that the inputted user is the User type
+      const firebaseUser = inputUser;
+      if (storageEngine instanceof FirebaseStorageEngine) {
+        const authInfo = await storageEngine?.getUserManagementData('authentication');
+        const isAuthEnabled = authInfo !== null ? authInfo.isEnabled : true;
```

Comment:
I think you also need to check undefined here, so we can just remove the explicit null check

```suggestion
        const isAuthEnabled = authInfo ? authInfo.isEnabled : true;
```